modules/core/src/encoder/SegmentEncoder.ts

```diff
@@ -77,16 +77,23 @@ export class SegmentEncoder {
 
   public static decode(encodedString: string, tcModel: TCModel, segment: string): TCModel {
 
-    const sequence = this.fieldSequence[''+tcModel.version][segment];
     const bitField = Base64Url.decode(encodedString);
     let bStringIdx = 0;
 
+    if (segment === Segment.CORE) {
+
+      tcModel.version = IntEncoder.decode(bitField.substr(bStringIdx, BitLength[Fields.version]), BitLength[Fields.version]);
+
+    }
+
     if (segment !== Segment.CORE) {
```

Comment:
Can we just make this an `else` to the `if` that you're adding since this is the other case?

Reply:
I thought about it.
However, there are two different case.
First case is detecting version.
Second case refers to segment type detecting.
Two "if" are better for future refactoring.

Reply:
Fair enough...